File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        str1=""
        str2=""
        while l1!=None:
            str1+=str(l1.val)
            l1=l1.next
        while l2!=None:
            str2+=str(l2.val)
            l2=l2.next
        
        str3=str(int(str1[::-1]) +int(str2[::-1]))
        i=len(str3)-1
        head=ListNode()
        temp=head
        while i>=0:
            temp1=ListNode(int(str3[i]))
            temp.next=temp1
            temp=temp.next
            i-=1
        return head.next

        

File: 2-add-two-numbers/add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* node=new ListNode(0);
        ListNode* temp=node;
        int carry =0;
        while(l1!=NULL || l2!=NULL){
            int sum=0;
            if((l1 && l2)){
                sum=l1->val+l2->val+carry;
                temp->next=new ListNode(sum%10);
                sum/=10;
                carry=sum%10;
                l1=l1->next;
                l2=l2->next;
            }else if(l1==NULL && l2!=NULL){
                sum=l2->val+carry;
                temp->next=new ListNode(sum%10);
                sum/=10;
                carry=sum%10;
                l2=l2->next;
            }else if(l2==NULL && l1!=NULL){
                sum=l1->val+carry;
                temp->next=new ListNode(sum%10);
                sum/=10;
                carry=sum%10;
                l1=l1->next;
            }
            temp=temp->next;
        }
        if(carry){
            temp->next=new ListNode(carry);
        }
        return node->next;
    }
};